addons/point_of_sale/models/pos_session.py

```diff
@@ -1384,7 +1384,7 @@ def _get_sale_vals(self, key, sale_vals):
             'balance': amount_converted,
         }
         if partial_vals.get('product_id'):
-            partial_vals['quantity'] = sale_vals.get('quantity') or 1
+            partial_vals['quantity'] = sale_vals.get('quantity', 1.00) * sign
```

Comment:
?

Reply:
This change optimizes the previous logic by setting the default quantity to 1.00 when it's not provided. 
This ensures every line has a valid quantity, and multiplying by sign adjusts for refunds or regular sales.